client/components/data/query-rewind-status/index.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { getRewindStatus } from 'state/activity-log/actions';
+import { isFetchingRewindStatus } from 'state/activity-log/selectors';
+
+class QueryRewindStatus extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired
+	};
+
+	componentWillMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.siteId !== nextProps.siteId ) {
+			this.request( nextProps );
+		}
+	}
+
+	request( props ) {
+		if ( props.requestingRewindStatus ) {
```

Comment:
Note that these `isFetching...` parts are not typically necessary, since the data-layer prevents re-sending of in-flight requests.